packages/shell-api/src/shell-api.ts

```diff
@@ -135,16 +136,38 @@ async function useCompleter(
   return await params.getDatabaseCompletions(args[1] ?? '');
 }
 
-/**
- * Complete a `show` subcommand.
- */
-// eslint-disable-next-line @typescript-eslint/require-await
-async function showCompleter(
-  params: ShellCommandAutocompleteParameters,
+// eslint-disable-next-line no-control-regex
+const CONTROL_CHAR_REGEXP = /[\x00-\x1F\x7F-\x9F]/g;
+
+async function newUseCompleter(
+  context: NewShellCommandAutocompleteParameters,
   args: string[]
 ): Promise<string[] | undefined> {
   if (args.length > 2) return undefined;
-  if (args[1] === 'd') {
+
+  let connectionId: string;
+
+  try {
+    ({ connectionId } = context.currentDatabaseAndConnection());
+  } catch (err: any) {
+    if (err.name === 'MongoshInvalidInputError') {
```

Comment:
I still really think this interface should not be try/catch based and instead allow for returning `undefined`, the fact that we now have to do this in multiple places does make me want to push for this a bit harder